tests/ClockMockTest.php

```diff
@@ -52,6 +52,37 @@ public function test_DateTimeImmutable_constructor_with_timezone()
         $this->assertEquals($dateWithTimezone, new \DateTimeImmutable('1986-06-05 14:41:32+02:00'));
     }
 
+    public function test_DateTimeImmutable_constructor_with_timezone_respect_zone_type()
+    {
+        ClockMock::freeze(new \DateTimeImmutable('now'));
+
+        $timezoneType3 = 'Asia/Tokyo';
+        $date = new \DateTimeImmutable('1986-06-05', new \DateTimeZone($timezoneType3));
+
+        $this->assertEquals($date->getTimezone()->getName(), $timezoneType3);
+
+        $timezoneType2 = 'CDT';
+        $date = new \DateTimeImmutable('1986-06-05', new \DateTimeZone($timezoneType2));
+
+        $this->assertEquals($date->getTimezone()->getName(), $timezoneType2);
+    }
+
+    public function test_DateTimeImmutable_constructor_without_timezone()
+    {
+        $originalTimezone = date_default_timezone_get();
+
+        $defaultTimezone = 'Asia/Tokyo';
+        date_default_timezone_set($defaultTimezone);
+
+        ClockMock::freeze(new \DateTimeImmutable('1986-06-05'));
+
+        $newDate = new \DateTimeImmutable('1986-06-05');
+
+        $this->assertEquals($newDate->getTimezone()->getName(), $defaultTimezone);
+
+        date_default_timezone_set($originalTimezone); // Revert timezone.
```

Comment:
Just a minor thing: if the above assertion fails, this line won't be executed and will cause failure of all the tests executed after this one. I would suggest to replace the last 2 lines with:
```
        try {
            $this->assertEquals($newDate->getTimezone()->getName(), $defaultTimezone);
        } finally {
            date_default_timezone_set($originalTimezone); // Revert timezone.
        }
```

Then we're good to go. Thanks!

Reply:
Yep makes sense. I'll update it.

Reply:
@asprega pardon the delay, requested change has been added.